zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -15,29 +15,173 @@ enum class Status {
+        else -> ValidationResult(Status.INVALID, "Vans power per charge point is outside the valid range (3..150 kW)")
+    }
+}
+
+// Validator for total charge point power < contracted capacity + battery power
+val validateTotalPowerChargePoints = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+
+    val totalPowerChargePoints = (gridConnection.transport.cars.powerPerChargePointKw ?: 0).toFloat() +
+            (gridConnection.transport.trucks.powerPerChargePointKw ?: 0).toFloat() +
+            (gridConnection.transport.vans.powerPerChargePointKw ?: 0).toFloat()
+
+    val contractedCapacity = (gridConnection.electricity.getContractedConnectionCapacityKw() ?: 0.0).toFloat()
+    val batteryPower = (gridConnection.storage.batteryPowerKw ?: 0.0).toFloat()
+
+    when {
+        totalPowerChargePoints < (contractedCapacity + batteryPower) -> ValidationResult(Status.VALID, "Total power of charge points is valid")
+        else -> ValidationResult(Status.INVALID, "Total power of charge points exceeds allowed capacity")
+    }
+}
+
+// Validator for vehicle travel distance in range 5k..100k km
+val validateCarTravelDistance = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    val travelDistanceCars = gridConnection.transport.cars.annualTravelDistancePerCarKm
+
+    when {
+        travelDistanceCars in 5000..100000 -> ValidationResult(Status.VALID, "Car travel distances are valid")
+        else -> ValidationResult(Status.INVALID, "Car travel distances are outside the valid range (5k..100k km)")
+    }
+}
+val validateTruckTravelDistance = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    val travelDistanceTrucks = gridConnection.transport.trucks.annualTravelDistancePerTruckKm
+
+    when {
+        travelDistanceTrucks in 5000..100000 -> ValidationResult(Status.VALID, "Truck travel distances are valid")
+        else -> ValidationResult(Status.INVALID, "Truck travel distances are outside the valid range (5k..100k km)")
+    }
+}
+val validateVanTravelDistance = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    val travelDistanceVans = gridConnection.transport.vans.annualTravelDistancePerVanKm
+
+    when {
+        travelDistanceVans in 5000..100000 -> ValidationResult(Status.VALID, "Van travel distances are valid")
+        else -> ValidationResult(Status.INVALID, "Van travel distances are outside the valid range (5k..100k km)")
+    }
+}
+
+
+// Validator for number of electric vehicles should be less than or equal to total number of vehicles
+val validateTotalElectricCars = Validator { survey: Survey ->
+    val gridConnection = survey.getSingleGridConnection()
+    when {
+        (gridConnection.transport.cars.numElectricCars ?: 0) > (gridConnection.transport.cars.numCars ?: 0) -> ValidationResult(Status.INVALID, "Electric Cars exceeds Number of Cars")
```

Comment:
For this and below:

* Watch capitalization
* Improve the sentences a bit, like "Number of electric cars exceeds the total number of cars"